sqlglot/optimizer/scope.py

```diff
@@ -587,7 +587,11 @@ def _traverse_tables(scope):
     for join in scope.expression.args.get("joins") or []:
         expressions.append(join.this)
 
-    expressions.extend(scope.expression.args.get("laterals") or [])
+    for lateral in scope.expression.args.get("laterals") or []:
+        if isinstance(lateral, exp.Join):
```

Comment:
i'm not sure i still understand this.

if you have

select ... lateral join y

you add the join into expressions but not the lateral? 

but why?

Reply:
As the lateral is nested within the join. As i mentioned earlier i would see this as a join on the lateral subquery or table valued function. Therefore the lateral is part of the "this" arg of the join, while the join is listed at the "joins".